File: test-npm-pip-pkg/ts/tests/test-chat-ai.ts
import { pathToFileURL } from 'url';
import { client, runTestSection, logResult } from "../utils/utils.js";

async function testChatAi() {
  console.log("\n🚀 Testing Chat AI Resource...");

  // 1. Assistants
  let testAssistantId: string | null = null;
  await runTestSection("chatAi.listAssistants", async () => {
    const list = await client.chatAi.listAssistants();
    logResult("Assistants", list);
  });

  await runTestSection("chatAi.createAssistant", async () => {
    const assistant = await client.chatAi.createAssistant({
      name: `SDK_UNIT_TEST_${Date.now()}`,
      workflow_name: `sdk_unit_test_${Date.now()}`,
      description: "Temporary assistant for unit testing",
    });
    testAssistantId = assistant.id;
    logResult("Created Assistant", assistant);
  });

  if (testAssistantId) {
    await runTestSection("chatAi.getAssistant", async () => {
      const assistant = await client.chatAi.getAssistant(testAssistantId!);
      logResult("Fetched Assistant", assistant);
    });

    await runTestSection("chatAi.updateAssistant", async () => {
        const updated = await client.chatAi.updateAssistant(testAssistantId!, {
            name: `SDK_UNIT_TEST_UPDATED_${Date.now()}`,
            workflow_name: `sdk_unit_test_${Date.now()}`,
        });
        logResult("Updated Assistant", updated);
    });
  }

  // 2. Models
  await runTestSection("chatAi.listModels", async () => {
    const models = await client.chatAi.listModels();
    logResult("Models", models);
  });

  // 3. Completions
  await runTestSection("chatAi.chat", async () => {
    const response = await client.chatAi.chat({
      model: "gpt-4o",
      messages: [{ role: "user", content: "Hello, say 'ready' if you hear me." }],
    });
    logResult("Chat Response", response.choices[0].message.content);
  });

  // 4. Folders & Knowledge
  let testFolderId: string | null = null;
  await runTestSection("chatAi.listFolders", async () => {
    const folders = await client.chatAi.listFolders();
    logResult("Folders", folders);
  });

  await runTestSection("chatAi.createFolder", async () => {
    const folder = await client.chatAi.createFolder({ name: `TEST_FOLDER_${Date.now()}` });
    testFolderId = folder.id || folder._id;
    logResult("Created Folder", folder);
  });

  // Cleanup
  if (testFolderId) {
    await runTestSection("chatAi.deleteFolder", async () => {
        await client.chatAi.deleteFolder(testFolderId!);
        console.log("   Folder deleted.");
    });
  }

  if (testAssistantId) {
    await runTestSection("chatAi.deleteAssistant", async () => {
      await client.chatAi.deleteAssistant(testAssistantId!);
      console.log("   Assistant deleted.");
    });
  }

  // 5. Prompts
  await runTestSection("chatAi.listPrompts", async () => {
    const prompts = await client.chatAi.listPrompts();
    logResult("Prompts Count", prompts.length);
  });

  // 6. Tools
  await runTestSection("chatAi.listTools", async () => {
    const tools = await client.chatAi.listTools();
    logResult("Tools Count", tools.length);
  });

  // 7. Knowledge Bases
  let testKnowledgeId: string | null = null;
  await runTestSection("chatAi.knowledgeLifecycle", async () => {
    // 1. List
    const kb = await client.chatAi.listKnowledge();
    logResult("Knowledge Bases Count", kb.length);

    // 2. Create (Direct)
    const newKb = await client.chatAi.createKnowledge({
        name: `SDK Direct KB ${Date.now()}`,
        description: "Created directly without folder"
    });
    testKnowledgeId = newKb.id;
    logResult("Created Knowledge", testKnowledgeId);

    // 3. Get
    const fetched = await client.chatAi.getKnowledge(testKnowledgeId!);
    logResult("Fetched Knowledge", fetched.name);

    // 4. Delete
    await client.chatAi.deleteKnowledge(testKnowledgeId!);
    console.log("   Knowledge deleted.");
  });

  // 8. Audio (Transcribe & Speech)
  await runTestSection("chatAi.speech", async () => {
    try {
        const res = await client.chatAi.speech({
            model: "tts-1",
            input: "Hello world",
            voice: "alloy"
        });
        logResult("Speech Response OK", res.ok);
    } catch (e) {
        console.warn("   [Skip] chatAi.speech failed");
    }
  });

  // 9. Document AI
  await runTestSection("chatAi.listDocumentModels", async () => {
    const models = await client.chatAi.listDocumentModels();
    logResult("Document Models Count", models.length);
  });

  await runTestSection("chatAi.processDocument", async () => {
    const orgId = process.env.IMBRACE_ORGANIZATION_ID || "";
    try {
        const res = await client.chatAi.processDocument({
            modelName: "gpt-4o",
            url: "https://example.com/invoice.pdf",
            organizationId: orgId,
            additionalInstructions: "Extract total amount"
        });
        logResult("Process Document", res.success);
    } catch (e) {
        console.warn("   [Skip] chatAi.processDocument failed (likely invalid URL/config)");
    }
  });

  console.log("\n✅ Chat AI Resource Testing Completed.");
  }


if (import.meta.url === pathToFileURL(process.argv[1]).href) {
  testChatAi().catch(console.error);
}

export { testChatAi };

File: test-local-pkg/py/tests/test_chat_ai.py
import time
from utils.utils import client, run_test_section, log_result, organization_id

def test_chat_ai():
    print("\n🚀 Testing Chat AI Resource...")

    state = {"assistant_id": None, "folder_id": None}

    # 1. Assistants
    def assistant_ops():
        list_res = client.chat_ai.list_assistants()
        log_result("Assistants", list_res)
        
        created = client.chat_ai.create_assistant({
            "name": f"SDK_PY_TEST_{int(time.time())}",
            "workflow_name": f"sdk_py_test_{int(time.time())}",
            "description": "Temp assistant",
        })
        state["assistant_id"] = created["id"]
        log_result("Created Assistant", state["assistant_id"])

        fetched = client.chat_ai.get_assistant(state["assistant_id"])
        log_result("Fetched Assistant", fetched)

        updated = client.chat_ai.update_assistant(state["assistant_id"], {
            "name": f"SDK_PY_TEST_UPDATED_{int(time.time())}",
            "workflow_name": f"sdk_py_test_{int(time.time())}",
        })
        log_result("Updated Assistant", updated)

        client.chat_ai.update_assistant_instructions(state["assistant_id"], "New instructions")
        log_result("Updated Instructions", True)

    run_test_section("Assistant CRUD", assistant_ops)

    # 2. Chat Sessions
    def chat_sessions():
        chats = client.chat_ai.list_chats()
        log_result("Chats Count", len(chats))

        new_chat = client.chat_ai.create_chat({"title": f"SDK Test Chat {int(time.time())}"})
        chat_id = new_chat.get("id") or new_chat.get("_id")
        log_result("Created Chat", chat_id)

        if chat_id:
            msgs = client.chat_ai.list_messages(chat_id)
            log_result("Messages Count", len(msgs))

            client.chat_ai.delete_chat(chat_id)
            log_result("Deleted Chat", True)
    run_test_section("Chat Sessions", chat_sessions)

    # 3. Models
    run_test_section("chat_ai.list_models", lambda: log_result("Models", client.chat_ai.list_models()))

    # 4. Completions
    def chat_completion():
        res = client.chat_ai.chat({
            "model": "gpt-4o",
            "messages": [{"role": "user", "content": "Hello, say 'py-ready' if you hear me."}]
        })
        log_result("Chat Response", res["choices"][0]["message"]["content"])
    run_test_section("chat_ai.chat", chat_completion)

    # 5. Folders & Knowledge Hubs
    def folder_ops():
        list_res = client.chat_ai.list_folders()
        log_result("Folders", list_res)

        kb_hubs = client.chat_ai.list_knowledge_hubs()
        log_result("Knowledge Hubs", len(kb_hubs))

        created = client.chat_ai.create_folder(f"TEST_PY_FOLDER_{int(time.time())}")
        state["folder_id"] = created.get("id") or created.get("_id")
        log_result("Created Folder", state["folder_id"])
    run_test_section("Folder & KB Ops", folder_ops)

    # Cleanup
    def cleanup():
        if state["folder_id"]:
            client.chat_ai.delete_folder(state["folder_id"])
            print("   Folder deleted.")
        if state["assistant_id"]:
            client.chat_ai.delete_assistant(state["assistant_id"])
            print("   Assistant deleted.")
    run_test_section("Cleanup", cleanup)

    # 6. Prompts & Tools
    run_test_section("chat_ai.list_prompts", lambda: log_result("Prompts", client.chat_ai.list_prompts()))
    run_test_section("chat_ai.list_tools", lambda: log_result("Tools", client.chat_ai.list_tools()))

    # 7. Document AI
    run_test_section("chat_ai.list_document_models", lambda: log_result("Document Models", client.chat_ai.list_document_models()))
    def test_process_doc():
        try:
            res = client.chat_ai.process_document(
                model_name="gpt-4o",
                url="https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf",
                organization_id=organization_id or "dummy_org"
            )
            log_result("Process Document", res)
        except Exception as e:
            print(f"   ⚠️ process_document failed: {str(e)}")
    run_test_section("chat_ai.process_document", test_process_doc)

    print("\n✅ Chat AI Resource Testing Completed.")

if __name__ == "__main__":
    test_chat_ai()
